custom_components/ttlock/services.py

```diff
@@ -153,3 +168,18 @@ async def handle_cleanup_passcodes(self, call: ServiceCall) -> ServiceResponse:
                     removed.append(code.name)
 
         return {"removed": removed}
+    
+    async def handle_set_autolock(self, call: ServiceCall):
+        """Set the autolock seconds"""
+
+        if call.data.get(CONF_ENABLED):
+            seconds = call.data.get(CONF_SECONDS) if call.data.get(CONF_SECONDS) > 0 else 10
+        else:
+            seconds = 0
+
+        for coordinator in self._get_coordinators(call):
```

Comment:
Probably need to check if the lock_id and the co-ordinator line up here.

Reply:
Again this I have no real idea what is happening, I just copied the other service. So I have no idea how to check that 🤦

Reply:
ah, never mind - I forgot this was wrapped up in `_get_coordinators(call)`

It works because call filters on `entity_id`